### PR TITLE
Add support for the prune endpoint

### PR DESCRIPTION
**Pull request by VagantemNumen**
Adds functions to get the amount of members that could be pruned
and to prune members using the prune endpoint.

### CODE REVIEW

**Comment by VagantemNumen** [comment]
I am fairly certain the account does not have proper permissions for ``TestGuildPrune()``.
It passes all the tests on my machine.
[discordgo_tests.log.txt](https://github.com/bwmarrin/discordgo/files/576828/discordgo_tests.log.txt)

File: restapi.go [hunk 1]
```diff
@@ -874,6 +874,79 @@ func (s *Session) GuildRoleDelete(guildID, roleID string) (err error) {
...
+		err = errors.New("The number of days should be more than or equal to 1.")
+		return
+	}
+
+	u, err := url.Parse(EndpointGuildPrune(guildID))
+	if err != nil {
+		return
+	}
+
+	q := u.Query()
```
Comment (iopred) [inline restapi.go]: Should be fine just to do:

uri := EndpointGuildPrune(guildID) + "?days="+days

Comment (VagantemNumen) [inline restapi.go]: Looked ugly and the ``net/url`` package was already imported anyway.
Should I change?

**Comment by VagantemNumen** [comment]
Done.

File: restapi.go [hunk 2]
```diff
@@ -874,6 +874,71 @@ func (s *Session) GuildRoleDelete(guildID, roleID string) (err error) {
...
+	if days <= 0 {
+		err = errors.New("The number of days should be more than or equal to 1.")
+		return
+	}
+
+	p := struct {
+		Pruned uint32 `json:"pruned"`
+	}{}
+
+	uri := EndpointGuildPrune(guildID) + "?days=" + fmt.Sprintf("%d", days)
```
Comment (iopred) [inline restapi.go]: Might be cleaner:

uri := EndpointGuildPrune(guildID) + fmt.Sprintf("?days="%d", days)

**Comment by iopred** [comment]
If the test is failing, it's probably better to leave it out :(

Otherwise we need @bwmarrin to change the testing bot permissions.

**Comment by VagantemNumen** [comment]
Okay, I will comment it out.

**Comment by VagantemNumen** [comment]
Though it is pretty weird that only one test is failing when the docs say they both require the same permissions.

**Pull request merged**